Detectors/MUON/MCH/Align/src/Aligner.cxx

```diff
@@ -351,24 +363,28 @@ void Aligner::init(TString DataRecFName, TString ConsRecFName)
 //_____________________________________________________
 void Aligner::terminate()
 {
-  mRecordWriter->terminate();
   fInitialized = kFALSE;
   LOG(info) << "Closing Evaluation TFile";
-  if (fTFile && fTTree) {
-    fTFile->cd();
-    fTTree->Write();
-    fTFile->Close();
+  if (fDoEvaluation) {
+    if (fTFile && fTTree) {
+      fTFile->cd();
+      fTTree->Write();
+      fTFile->Close();
+    }
   }
+  delete mRecordWriter;
+  delete mRecordReader;
+  delete fMillepede;
+  delete fTrackRecord;
```

Comment:
These deletes should go in the destructor.
Actually, since these pointers point to objects produced only once in the constructor, you could use data member objects instead of pointers. If you don't want to change the others, at least `fTrackRecord` can remain a data member object as before.